testing-common/src/main/java/io/opentelemetry/javaagent/testing/common/AgentTestingExporterAccess.java

```diff
@@ -0,0 +1,277 @@
+      Resource resource = resourceSpans.getResource();
+      for (InstrumentationLibrarySpans ilSpans :
+          resourceSpans.getInstrumentationLibrarySpansList()) {
+        InstrumentationLibrary instrumentationLibrary = ilSpans.getInstrumentationLibrary();
+        for (Span span : ilSpans.getSpansList()) {
+          spans.add(
+              TestSpanData.builder()
+                  .setTraceId(TraceId.bytesToHex(span.getTraceId().toByteArray()))
+                  .setSpanId(SpanId.bytesToHex(span.getSpanId().toByteArray()))
+                  .setTraceState(extractTraceState(span.getTraceState()))
+                  .setParentSpanId(SpanId.bytesToHex(span.getParentSpanId().toByteArray()))
+                  .setResource(
+                      io.opentelemetry.sdk.resources.Resource.create(
+                          fromProto(resource.getAttributesList())))
+                  .setInstrumentationLibraryInfo(
+                      InstrumentationLibraryInfo.create(
+                          instrumentationLibrary.getName(), instrumentationLibrary.getVersion()))
+                  .setName(span.getName())
+                  .setStartEpochNanos(span.getStartTimeUnixNano())
+                  .setEndEpochNanos(span.getEndTimeUnixNano())
+                  .setAttributes(fromProto(span.getAttributesList()))
+                  .setEvents(
+                      span.getEventsList().stream()
+                          .map(
+                              event ->
+                                  SpanData.Event.create(
+                                      event.getTimeUnixNano(),
+                                      event.getName(),
+                                      fromProto(event.getAttributesList()),
+                                      event.getDroppedAttributesCount()
+                                          + event.getAttributesCount()))
+                          .collect(Collectors.toList()))
+                  .setStatus(fromProto(span.getStatus()))
+                  .setKind(fromProto(span.getKind()))
+                  .setLinks(
+                      span.getLinksList().stream()
+                          .map(
+                              link ->
+                                  SpanData.Link.create(
+                                      SpanContext.create(
+                                          TraceId.bytesToHex(link.getTraceId().toByteArray()),
+                                          SpanId.bytesToHex(link.getSpanId().toByteArray()),
+                                          TraceFlags.getDefault(),
+                                          extractTraceState(link.getTraceState())),
+                                      fromProto(link.getAttributesList()),
+                                      link.getDroppedAttributesCount() + link.getAttributesCount()))
+                          .collect(Collectors.toList()))
+                  // OTLP doesn't have hasRemoteParent
```

Comment:
I realized this is why a lot of people have problems identifying local root in collector :O